test/multiplexed-session.ts

```diff
@@ -0,0 +1,326 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {beforeEach, afterEach, describe, it} from 'mocha';
+import * as events from 'events';
+import * as sinon from 'sinon';
+import {grpc, Status} from 'google-gax';
+import {Database} from '../src/database';
+import {Session} from '../src/session';
+import {MultiplexedSession} from '../src/multiplexed-session';
+import {Transaction} from '../src/transaction';
+
+class FakeTransaction {
+  options;
+  constructor(options?) {
+    this.options = options;
+  }
+  async begin(): Promise<void> {}
+}
+
+describe('MultiplexedSession', () => {
+  let multiplexedSession;
+
+  function noop() {}
+  const DATABASE = {
+    createSession: noop,
+    databaseRole: 'parent_role',
+  } as unknown as Database;
+
+  let fakeMuxSession;
+  let createSessionStub;
+  const sandbox = sinon.createSandbox();
+
+  const createSession = (name = 'id', props?): Session => {
+    props = props || {};
+
+    return Object.assign(new Session(DATABASE, name), props, {
+      create: sandbox.stub().resolves(),
+      transaction: sandbox.stub().returns(new FakeTransaction()),
+      getMetadata: sandbox.stub().resolves({multiplexed: true}),
+    });
+  };
+
+  const shouldNotBeCalled = sandbox.stub().throws('Should not be called.');
+
+  beforeEach(() => {
+    fakeMuxSession = createSession();
+    createSessionStub = sandbox
+      .stub(DATABASE, 'createSession')
+      .withArgs({multiplexed: true})
+      .callsFake(() => {
+        return Promise.resolve([fakeMuxSession]);
+      });
+    multiplexedSession = new MultiplexedSession(DATABASE);
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  describe('instantiation', () => {
+    it('should localize the database instance', () => {
+      assert.strictEqual(multiplexedSession.database, DATABASE);
+    });
+
+    it('should apply defaults', () => {
+      assert.strictEqual(multiplexedSession.refreshRate, 7);
+    });
+
+    it('should create a session object', () => {
+      assert.deepStrictEqual(multiplexedSession._multiplexedSession, null);
+    });
+
+    it('should inherit from EventEmitter', () => {
+      assert(multiplexedSession instanceof events.EventEmitter);
+    });
+  });
+
+  describe('createSession', () => {
+    let _createSessionStub;
+    let _maintainStub;
+
+    beforeEach(() => {
+      _maintainStub = sandbox.stub(multiplexedSession, '_maintain');
+      _createSessionStub = sandbox
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+    });
+
+    it('should create mux session', () => {
+      multiplexedSession.createSession();
+      assert.strictEqual(_createSessionStub.callCount, 1);
+    });
```

Comment:
You could stub `_createSession` to return a session not only resolves and verify the multiplexedSession property.

Reply:
there is already a test where we are verifying the multiplexed property for the session
`should have the multiplexed property set to true`